### ISSUE TITLE
Question: bug with style-loader?

### ISSUE THREAD

**Issue by Bogdaan** [kriasoft/react-starter-kit#375]
When i work with css background-image source map generate "Blob" url, and images not loaded from dev-server:
  
    Request URL:blob:http%3A//localhost%3A3000/2454cd1b-a4bb-46fa-8d4b-446ac9b2c9a6   

In style-loader i found this note: [output.publicPath](https://github.com/webpack/style-loader/issues/55)

It posible load image from files with current webpack config (with publicPath=/)?

**Comment by pixelcure** [comment]
I'm also experiencing this issue...

**Comment by thoughtassassin** [comment]
I am experiencing this issue with fonts that are loaded in the public folder. They are visible on localhost:5000 but on localhost:3000 the fonts show for a fraction of a second and then the styles and the fonts disappear.

**Comment by nss213** [comment]
I just specified the publicPath for the client to work around this. I _believe_ it only shows up in dev mode.

```javascript
const clientConfig = merge({}, config, {
  entry: [
    ...(WATCH ? ['webpack-hot-middleware/client'] : []),
    './src/client.js',
  ],
  output: {
    path: path.join(__dirname, '../build/public'),

    // need to give absolute URL when using sourcemaps/blobs
    // https://github.com/webpack/css-loader/issues/29
    // https://github.com/webpack/style-loader/issues/55
    publicPath: DEBUG ? 'http://localhost:3000/': '/',

    filename: DEBUG ? '[name].js?[hash]' : '[name].[hash].js',
  },
```

**Comment by Bogdaan** [comment]
@nss213 thanks, this also works

**Comment by Ilshidur** [comment]
@nss213 Worked for me :
```javascript
const clientConfig = merge({}, config, {
  entry: './src/client.js',
  output: {
    path: path.join(__dirname, '../build/public'),
    publicPath: DEBUG ? 'http://localhost:3000/' : '/',
    filename: DEBUG ? '[name].js?[hash]' : '[name].[hash].js',
  },
```

**Comment by arao456** [comment]
@nss213 @Ilshidur 

Did this break connecting to dev from a mobile device for you guys? When I try to connect to 192.168.....:3000 my mobile console is showing "ERR_CONNECTION_REFUSED" when trying to fetch main.js, but when I take out the public path line all is well.

**Comment by Bogdaan** [comment]
@arao456 no, browsersync works fine. But you can try change url-loader to

    'url-loader?limit='+(DEBUG? 9999999: 1000),

**Comment by arao456** [comment]
@Bogdaan that seems to work for me. What also works is setting publicPath to `publicPath: DEBUG ? 'http://<192.168...>:3000/' : '/'`, rather than `localhost` which leads me to think that perhaps my phone or other external device is trying to load assets from somewhere it doesn't have access too (i.e. it's own localhost?).

**Comment by Bogdaan** [comment]
@arao456  yes, of course if you devices are connected via a network - you need external address

**Comment by arao456** [comment]
@Bogdaan I realize this now, but I guess the point I'm getting at is that the fix using `publicPath` isn't quite a permanent one so maybe we should look into an alternative? Either way, your fix works, so thanks!
